### ISSUE TITLE
PGP public key database

### ISSUE THREAD

**Issue by ion1** [ipfs/archives#27]
It would be nice to have the PGP public key database in IPFS.

[SKS (OpenPGP keyserver)](https://bitbucket.org/skskeyserver/sks-keyserver/wiki/Home)

**Comment by davidar** [comment]
:+1: https://pgp.key-server.io/sks-dump/

**Comment by davidar** [comment]
https://ipfs.io/ipfs/Qmbs2DxMBraF3U8F7vLAarGmZaSFry3vVY5zytuN3BxwaY

- [ ] connect with https://github.com/ipfs/specs/tree/master/keychain

**Comment by simonv3** [comment]
Just putting keybase.io here: 

https://keybase.io/

**Comment by davidar** [comment]
@simonv3 https://keybase.io/davidar ;)

**Comment by simonv3** [comment]
@davidar Heh, sadly, I don't have a keybase account, just thought it was a relevant source, maybe to talk to them as well in the future.

**Comment by davidar** [comment]
@simonv3 definitely, they even [publish their data](https://keybase.io/docs/server_security) as a Merkle Tree, so it would probably fit in quite nicely with IPFS.

CC: @jbenet @keybase @lord @Sidnicious

PS: @simonv3 send me an email if you want an invite

**Comment by jbenet** [comment]
> they even publish their data as a Merkle Tree, so it would probably fit in quite nicely with IPFS.

Nice! So if we ingest it cleverly, we can FUSE mount keybase with ipfs.

```
cat /ipns/data.keybase.io/jbenet/public-key
```

**Comment by markg85** [comment]
Just curious, is there an update on this?
I'd like to fetch a public key from someone else via IPFS.

I guess what i'm asking is if there is a [SKS Keyserver](https://bitbucket.org/skskeyserver/sks-keyserver/wiki/Home) implementation to be used within IPFS? That would be so cool and would make another small part fully distributed!

Having said that, i'd expect it to be some sub-command of ipfs to interact with an "sks keyserver" (lets say it's called sks, or call it gpg). I'd then guess a command like: `ipfs sks get <key id>` to get me the details of that id. This would be the same as `gpg --recv-keys <id>`, Searching for one would be `ipfs sks get <key id>` which in the gpg command would be `gpg --search <id>`
An `id` can then be:
* `AFAFAFAFAFAFAFAF` (16 characters, the upper 64 bit of the fingerprint)
* `AFAFAFAFAFAFAFAFAFAFAFAFAFAFAFAF` (the full fingerprint
* `some@mail.com` (by email)
